vpr/src/route/rr_graph_uxsdcxx_serializer.h

```diff
@@ -1886,7 +1882,7 @@ class RrGraphSerializer final : public uxsd::RrGraphBase<RrGraphContextTypes> {
 
     const size_t num_arch_switches_;
     const t_arch_switch_inf* arch_switch_inf_;
-    const vtr::vector<RRSegmentId, t_segment_inf>& segment_inf_;
+    const vtr::vector<RRSegmentId, t_segment_inf> segment_inf_;
```

Comment:
We can keep the reference here. No need to remove the ``&``

Reply:
If we keep here & then we cannot this line `segment_inf_(rr_graph_builder->rr_segments())` remove from constructor .

Reply:
Understood. If the object requires a read-only copy, it should get from the ``RRGraphView`` but not ``RRGraphBuilder``. So it means, here we should use something like

```
segment_inf_(rr_graph->rr_segments())
```

Reply:
@tangxifan 
Ok let me see this way.